client/matchmaking/socket-handlers.ts

```diff
@@ -163,57 +167,60 @@ const eventToAction = {
     } = event.chosenMap
     // Even though we're downloading the whole map pool as soon as the player enters the queue,
     // we're still leaving this as a check to make sure the map exists before starting a game.
-    ipcRenderer.invoke('mapStoreDownloadMap', hash, format, mapUrl).catch(err => {
+    ipcRenderer.invoke('mapStoreDownloadMap', hash, format, mapUrl!)?.catch(err => {
       // TODO(tec27): Report this to the server so the loading is canceled immediately
 
       // This is already logged to our file by the map store, so we just log it to the console for
       // easy visibility during development
       console.error('Error downloading map: ' + err + '\n' + err.stack)
     })
 
-    const config = {
-      localUser: user.toJS(),
+    const config: GameLaunchConfig = {
+      localUser: {
+        id: user.id,
+        name: user.name,
+      },
       setup: {
         gameId: event.setup.gameId,
         name: 'Matchmaking game', // Does this even matter for anything?
         map: event.chosenMap,
-        gameType: 'oneVOne',
-        slots: event.slots,
-        host: event.slots[0], // Arbitrarily set first player as host
+        gameType: GameType.OneVsOne,
+        slots: event.slots as PlayerInfo[],
```

Comment:
extremely not a fan of these casts, if you're doing this either A) `event` is typed badly, or B) you're assuming that the thing that is being passed in the event will always have the same fields as the thing you're casting to. There's zero reason to assume that (and indeed, if the fields were exactly the same, this cast would be unnecessary in the first place), you should be creating an object of the correct type and mapping the fields manually